pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -433,12 +445,14 @@ func (nc *NodeController) deletePods(nodeID string) error {
 	if err != nil {
 		return err
 	}
+	nc.recordNodeEvent(nodeID, fmt.Sprintf("Deleting all Pods from Node %v.", nodeID))
 	for _, pod := range pods.Items {
 		// Defensive check, also needed for tests.
 		if pod.Spec.NodeName != nodeID {
 			continue
 		}
 		glog.V(2).Infof("Delete pod %v", pod.Name)
+		nc.recorder.Eventf(&pod, "NodeControllerEviction", "Deleting Pod %s from Node %s", pod.Name, nodeID)
```

Comment:
The message should be something like "Deleting pod from Node %s" because the pod name is already printed out when you print the event with `kubectl get events`, and it is implied when you do `kubectl describe pod foo`